.circleci/config.yml

```diff
@@ -288,26 +293,19 @@ commands:
           condition: << parameters.set_tarball_path >>
           steps:
             - run:
-                name: Set HERMES_ENGINE_TARBALL_PATH if present
+                name: Set HERMES_ENGINE_TARBALL_PATH envvar if Hermes tarball is present
                 command: |
-                  BASE_PATH=/tmp/hermes/hermes-runtime-darwin/
-                  if [ ! -d $BASE_PATH ]; then
-                    echo "Hermes tarball base path not present ($BASE_PATH). Build it from source."
-                    exit 0
-                  fi
-                  if [[ << parameters.flavor >> == "Debug" ]]; then
-                    TARBALL_FILENAME=hermes-runtime-darwin-debug-*.tar.gz
-                  elif [[ << parameters.flavor >> == "Release" ]]; then
-                    TARBALL_FILENAME=hermes-runtime-darwin-release-*.tar.gz
-                  else
-                    echo "Unsupported build type << parameters.flavor >>."
+                  HERMES_TARBALL_ARTIFACTS_DIR=<< parameters.hermes_tarball_artifacts_dir >>
+                  if [ ! -d $HERMES_TARBALL_ARTIFACTS_DIR ]; then
+                    echo "Hermes tarball artifacts dir not present ($HERMES_TARBALL_ARTIFACTS_DIR). Build Hermes from source."
                     exit 0
                   fi
-                  # /tmp/hermes/hermes-runtime-darwin/hermes-runtime-darwin-release-v0.70.0.tar.gz
-                  # /tmp/hermes/hermes-runtime-darwin/hermes-runtime-darwin-debug-v0.70.0.tar.gz
-                  TARBALL_PATH=$(ls $BASE_PATH$TARBALL_FILENAME)
+
+                  TARBALL_FILENAME=$(node ~/react-native/scripts/hermes/get-tarball-name.js -b << parameters.flavor >> -v *)
```

Comment:
The Circle CI job, before these changes, uses a `hermes-runtime-darwin-debug-*.tar.gz` glob to match *any* tarball in the artifacts directory:

```
if [[ << parameters.flavor >> == "Debug" ]]; then
  TARBALL_FILENAME=hermes-runtime-darwin-debug-*.tar.gz
elif [[ << parameters.flavor >> == "Release" ]]; then
  TARBALL_FILENAME=hermes-runtime-darwin-release-*.tar.gz
else
  echo "Unsupported build type << parameters.flavor >>."
  exit 0
fi
```

We do this because this is a command that can be reused and we do not know the version number for the tarball. While we could use a parameter to be specific, there is an assumption that the artifacts directory will contain at most one debug and one release tarball for an arbitrary React Native version (v1000.0.0 in most cases, or a RN version tag in stable releases).

I'm using the get-tarball-name script to standardize how the tarball filename is obtained instead of relying on magic strings throughout the code, so in this case passing a `*` version number should do the trick as long as $TARBALL_FILENAME is not used inside quotes.